lasagne/layers/recurrent.py

```diff
@@ -757,3 +762,306 @@ def step_masked(input_dot_W_n, mask_n, cell_previous, hid_previous,
         self.cell_out = cell_out
 
         return hid_out
+
+
+class GRULayer(Layer):
+    """Gated Recurrent Layer (GRU)
+
+    Layer with gated recurrent units implemented as in [1]_, [2]_.
```

Comment:
I would like to do this:

```Python
    Notes
    -----
    The GRU units is implemented slightly differently in [1]_ and in [2]_.
    We use the following equation to calculate the hidden update:

    .. math:: \hat{h}^j_t = tanh(W\mathbf{x}_t + \mathbf{r}_t \odot (U\mathbf{h_{t-1}}))^j

    An alternative formulation is:

    .. math:: \hat{h}^j_t = tanh(W\mathbf{x}_t + \odot U(\mathbf{r}_t\mathbf{h_{t-1}}))^j

    We use the first formulation because it allows us to do all matrix
    operations in a single dot product.
```

Do you know howi break the math in the doc string its too wide?